test/test_nn.py

```diff
@@ -11928,55 +11936,60 @@ def _test_EmbeddingBag(self, device, mode, sparse, dtype=torch.double, test_back
             self.assertRaises(RuntimeError, lambda: es(input.view(-1), offset))
 
     @dtypesIfCUDA(torch.half, torch.float, torch.double)
-    @dtypes(torch.float, torch.double)
+    @dtypes(torch.int, torch.long)
     def test_embedding_bag_device(self, device, dtype):
-        self._test_EmbeddingBag(device, 'sum', False, dtype)
-        self._test_EmbeddingBag(device, 'mean', False, dtype)
-        self._test_EmbeddingBag(device, 'max', False, dtype)
-
-        test_backward = False
-        if self.device_type == 'cuda':
-            # see 'todo' in test_embedding_bag.
-            test_backward = dtype is not torch.float16
-        elif self.device_type == 'cpu':
-            # TODO: figure out why precision on sparse embeddings isn't the
-            # same as for dense.
-            test_backward = dtype is not torch.float
-
-        self._test_EmbeddingBag(device, 'sum', True, dtype, test_backward=test_backward)
-        self._test_EmbeddingBag(device, 'mean', True, dtype, test_backward=test_backward)
+        wdtypes = [torch.float, torch.double]
+        for wdtype in wdtypes:
+            self._test_EmbeddingBag(device, 'sum', False, wdtype, dtype)
+            self._test_EmbeddingBag(device, 'mean', False, wdtype, dtype)
+            self._test_EmbeddingBag(device, 'max', False, wdtype, dtype)
+
+            test_backward = False
+            if self.device_type == 'cuda':
+                # see 'todo' in test_embedding_bag.
+                test_backward = wdtype is not torch.float16
+            elif self.device_type == 'cpu':
+                # TODO: figure out why precision on sparse embeddings isn't the
+                # same as for dense.
+                test_backward = wdtype is not torch.float
+
+            self._test_EmbeddingBag(device, 'sum', True, wdtype, dtype, test_backward=test_backward)
+            self._test_EmbeddingBag(device, 'mean', True, wdtype, dtype, test_backward=test_backward)
 
     @dtypesIfCUDA(torch.half, torch.float, torch.double)
```

Comment:
this does not look right, on cuda dtype will be half, float and double instead of integral. Same comment as above for specifying tuples of dtypes

Reply:
Thanks for the catch. Will fix.